docs/payloads.md

```diff
@@ -1,36 +1,36 @@
 # Document Payloads
 
-Usually, redisearch stores documents as HASH keys. But if you want to access some data for 
-aggregation or scoring functions, we might want to store that data as an inline payload. 
-This will allow us to evaluate properties of a document for scoring purposes at very low cost.
+Usually, RediSearch stores documents as HASH keys. But if you want to access some data for 
+aggregation or scoring functions, you might want to store that data as an inline payload. 
+This will allow you to evaluate properties of a document for scoring purposes, at a very low cost.
 
-Since the scoring functions already have access to the DocumentMetaData, which contains document flags and score,
-We can add custom payloads that can be evaluated in run-time.
+Since the scoring functions already have access to DocumentMetaData, which contains the document's flags and score,
+you can add custom payloads that can be evaluated at runtime.
 
-Payloads are NOT indexed and are not treated by the engine in any way. They are simply there for the purpose 
+Payloads are NOT indexed and are not treated by the engine in any way; they are simply there for the purpose 
 of evaluating them in query time, and optionally retrieving them. They can be JSON objects, strings, or preferably, 
-if you are interested in fast evaluation, some sort of binary encoded data which is fast to decode.
+if you are interested in fast evaluation, some sort of binary-encoded data (which is fast to decode).
 
-## Adding payloads for documents:
+## Adding Payloads for Documents
 
-When inserting a document using FT.ADD, you can ask RediSearch to store an arbitrary binary safe string as the document payload.
-This is done with the PAYLOAD keyword:
+When inserting a document using FT.ADD, you can ask RediSearch to store an arbitrary binary-safe string as the document payload--
+just use the PAYLOAD keyword:
 
 ```
 FT.ADD {index_name} {doc_id} {score} PAYLOAD {payload} FIELDS {field} {data}...
 ```
 
 ## Evaluating Payloads in Query Time
 
-When imlplementing a scoring function, the signature of the function exposed is:
+When implementing a scoring function, the signature of the function exposed is:
 
 ```c
 double (*ScoringFunction)(DocumentMetadata *dmd, IndexResult *h);
 ```
 
-> NOTE: currently scoring functions cannot be dynamically added, and forking the engine and replacing them is required.
+> NOTE: currently, scoring functions cannot be dynamically added; forking the engine and replacing them is required.
 
-DocumentMetaData includes a few fields, one of them being the payload. It wraps a simple byte array with
+DocumentMetaData includes a few fields, one of which is the payload. It wraps a simple byte array with
```

Comment:
It would be great if you could add an example of how to modify the scoring function using the payload. Thanks in advance!

Reply:
@vruizext have you looked at the examples of the extension API? There's no direct example, but there's an example scoring function, which can access the payload. All it needs to do is return a double. Whatever you do with the payload is up to you. 

Ok, here's a really stupid example. Suppose your payload is encoded doubles (not sure why, but it's the simplest example). You'd do:

```c
double myScorer(RSScoringFunctionCtx *ctx, RSIndexResult *h, RSDocumentMetadata *dmd,
                double minScore) {

  if (dmd->payload != NULL && dmd->payload->data  != NULL && dmd->payload->len == sizeof(double)) {
    return *(double *)dmd->payload->data;
  }

  return 0;
}
```

Reply:
Thanks @dvirsky, I've managed to build my own scoring function doing some changes to the default TF-IDF scorer, but couldn't figure out how to use the payload.